Builder error caused by variable name of array

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Pop!_OS 22.04 LTS
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
CC version: cc (Ubuntu 11.2.0-19ubuntu1) 11.2.0

getwd: /home/evgen/Desktop/playground
vmodules: /home/evgen/.vmodules
vroot: /home/evgen/Desktop/v
vexe: /home/evgen/Desktop/v/v
vexe mtime: 2022-10-23 11:14:40
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 721dbec.3f03520

Git version: git version 2.34.1
Git vroot status: weekly.2022.42-32-g3f035205
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg playground.v`
```v
module main

fn main() {
	array := []int{len: 4, init: it}
	println(array)
}
```

**What did you expect to see?**

[0, 1, 2, 3]


**What did you see instead?**
```
==================
/tmp/v_1000/playground.10460106522849273426.tmp.c:12105: error: lvalue expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```
**Additional Information**
It looks like if i literally name a variable "array" (other names don't seem to cause errors) and initialize it like this, i get a builder error

Also there would be another error if I make array fixed-size

**Code:**
```
module main

fn main() {
	array := [3]int{init: it}
	println(array)
}
```
**Error:**
```
==================
/tmp/v_1000/playground.9436549357264114873.tmp.c:12103: error: cannot convert 'typedef struct array' to 'int *'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.
...
```

--- a/vlib/builtin/array_test.v
+++ b/vlib/builtin/array_test.v
@@ -1601,3 +1601,9 @@ fn test_2d_array_init_with_it() {
 	a := [][]int{len: 6, init: f(it, 2 * it)}
 	assert a == [[0, 0], [1, 2], [2, 4], [3, 6], [4, 8], [5, 10]]
 }
+
+fn test_using_array_name_variable() {
+	array := []int{len: 4, init: it}
+	println(array)
+	assert array == [0, 1, 2, 3]
+}

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -448,7 +448,7 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 					g.is_shared = var_type.has_flag(.shared_f)
 					if is_fixed_array_init && !has_val {
 						if val is ast.ArrayInit {
-							g.array_init(val, ident.name)
+							g.array_init(val, c_name(ident.name))
 						} else {
 							g.write('{0}')
 						}
@@ -460,7 +460,7 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 							g.write('*')
 						}
 						if val is ast.ArrayInit {
-							g.array_init(val, ident.name)
+							g.array_init(val, c_name(ident.name))
 						} else if val_type.has_flag(.shared_f) {
 							g.expr_with_cast(val, val_type, var_type)
 						} else {

